File: backend/entity/modrating.go
package entity

import "gorm.io/gorm"

type ModRating struct {
	gorm.Model

	Score   int    `json:"score"`
	Comment string `json:"comment,omitempty" gorm:"type:text"`
	UserID  uint   `json:"user_id" gorm:"index"`
	User    *User  `gorm:"foreignKey:UserID" json:"user,omitempty"`
	ModID   uint   `json:"mod_id" gorm:"index"`
	Mod     *Mod   `gorm:"foreignKey:ModID" json:"-"`
}

File: frontend/src/interfaces/ModRating.ts
export interface ModRating {
  ID: number;
  score: number;
  user_id: number;
  mod_id: number;
  comment?: string;
}

export interface CreateModRatingRequest {
  score: number;
  user_id: number;
  mod_id: number;
  comment?: string;
}

export interface UpdateModRatingRequest {
  ID: number;
  score?: number;
  user_id?: number;
  mod_id?: number;
  comment?: string;
}
